### DISCUSSION TITLE
How to log StandardScaler along with model

### DISCUSSION THREAD

**Discussion by muthumula19** [mlflow/mlflow#5589] [Q&A] [answered]
I am new to mlflow , i log the KNN model successfully but i don't how to log  StandardScaler along with model, i need StandardScaler along with KNN to perform inverse transformation in the at end of prediction,

**Comment by BenWilson2** [comment]
For PySpark? You might want to try to wrap those Estimators up in a Pipeline. That way you don't have to mess about with figuring out how to get inference to work by fetching different artifacts.

**Comment by sniafas** [comment]
Does it work for you to pickle the Scaler object and save it as an artifact?

**Comment by JHibbard** [comment, marked as answer]
Hi muthumula19,

This workflow is now supported by MLflow Recipe's transformer step. [[here](https://fukumaruuu.medium.com/a-step-by-step-guide-to-using-mlflow-recipes-to-refactor-messy-notebooks-f78196b97ff0)] is a great blog showing how to refactor your code to take advantage of the new feature and [[here](https://www.mlflow.org/docs/latest/python_api/mlflow.recipes.html#module-mlflow.recipes.classification.v1.recipe)] is the documentation page on MLflow.

Hope that helps!
Jim

**Comment by jhammarstedt** [comment]
Is it possible to save the transformation step/scaler without using recipe?
